cli/src/diff_util.rs

```diff
@@ -114,7 +114,8 @@ fn diff_formats_from_args(
     .collect_vec();
     if let Some(name) = &args.tool {
         let tool = merge_tools::get_tool_config(settings, name)?
-            .unwrap_or_else(|| MergeTool::with_program(name));
+            .unwrap_or_else(|| ExternalMergeTool::with_program(name));
+        let tool = MergeTool::External(tool);
```

Comment:
Shouldn't it be `DiffFormat::Tool(Box<ExternalMergeTool>)`?

`DiffFormat` is the enum to select "internal" or "external" diff function.

Reply:
@yuja If we want to render diffs using scm-record in addition to possibly an external mergetool, how should we organize it? Do you think it's better to have another enum case in `DiffFormat`? I haven't implemented diffing via scm-record in jj yet, so I don't know what would be better. (I implemented it as a difftool via git-branchless, but it doesn't have internal diffing, so not the same.)

Reply:
I understand now: in this case, the diff tool is used to generate the input, rather than consume the output of something. I'll adjust it to be `ExternalMergeTool` instead of `MergeTool`.

Reply:
> the diff tool is used to generate the input, rather than consume the output of something.

Yes, it's not like [delta](https://github.com/dandavison/delta), but a `diff -uNr` command.

And it can't be curses UI because stdout is piped. GUI tool works, but I'm not sure if `jj diff --tool` should be overloaded in that way.